packages/dendron-plugin-views/src/components/DendronApp.tsx

```diff
@@ -178,7 +179,29 @@ function DendronVSCodeApp({ Component }: { Component: DendronComponent }) {
     }
   });
 
-  return <Component {...props} />;
+  const isLocked = props.ide.isLocked;
+
+  const handleLock = React.useCallback(() => {
```

Comment:
Why are you using `useCallback` here?

Reply:
I that that you just copied this from the Preview component, but I think just defining a simple non wrapped function (yes, that gets redefined every render) might be more apt.

https://kentcdodds.com/blog/usememo-and-usecallback

TLDR, this function call is neither expensive nor makes an expensive calculation.

Reply:
Thank you for that pointer. I had it previously inside `DendronNotePreview` thinking about "Referential equality" but also there I oversaw its uselessness. Makes me reflect on unconsicous assumptions or automatic responses I find myself inhabiting. [`1834f16` (#3293)](https://github.com/dendronhq/dendron/pull/3293/commits/1834f167e703eb4b02d1da67432a5c5f3f8a8172)

Reply:
I love that you linked the commit you made the change on.  Wow!  I've never seen that before and it's so smart.